core/controllers/profile.py

```diff
@@ -392,3 +392,32 @@ def get(self):
             'username': user_settings.username,
             'user_is_logged_in': True
         })
+
+
+class LoginUrlHandlerPage(base.BaseHandler):
+
+    @acl_decorators.open_access
+    def get(self):
+        login_url = None
+        if self.user_id:
+            login_url = None
+            self.render_json({'login_url': login_url})
+        else:
+            if self.request.GET.items():
+                if self.request.GET.items()[0]:
```

Comment:
You can use `self.request.get('current_url')` instead, it will simplify things a lot

Reply:
@YashJipkate What about this?